[!!!][TASK] Cache fully build NodeType configurations in production

kitsunet opened the pull request:
Breaking if you rely on post processors to be applied on each request.

aertmann commented on TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/NodeType.php:
```diff
@@ -371,7 +376,7 @@ public function getConfiguration($configurationPath)
     {
         $this->initialize();
-        return isset($this->fullConfiguration['ui']['label']) ? $this->fullConfiguration['ui']['label'] : '';
+        return isset($this->fullConfiguration['ui']) && isset($this->fullConfiguration['ui']['label']) ? $this->fullConfiguration['ui']['label'] : '';
```
why change this?

kitsunet replied:
Merge error, will change back.

aertmann replied:
still relevant?

kitsunet replied:
fixed.

aertmann commented on TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/NodeType.php:
```diff
@@ -125,6 +125,11 @@ public function __construct($name, array $declaredSuperTypes, array $configurati
 
         $this->localConfiguration = $configuration;
+
+        $this->fullConfiguration = $fullConfiguration;
```
seems odd setting it to null when there's a check it could be in if you ask me, but not important ofc.

kitsunet replied:
yeah, will recheck it again. The idea setting it to null was to save memory as the config is no longer needed when the node types are built.

aertmann replied:
ah I see, the default value is array not null.. makes sense, sorry

aertmann commented:
ah sorry, overlooked the ``WIP`` label.. anyway a good reason for allowing this potentially breaking change is that post processors are applied and cached already (client side), so I really doubt this will affect anyone since they would already have the problem..

kitsunet commented:
Ah, good to know. @dfeyer commented about the problem in the original change. I think we just need to try if it actually breaks something that isn't already behaving like that.

dfeyer commented:
Agree with @aertmann we try to use processor to get informations from external website ... but fail because of the client cache, now we use custom Inspector editor to connect to external data provider.

So for me it's fine if can merge this one.

kitsunet commented:
Rebased and fixed test problems, additionally moved the responsibility of this cache only being enabled in production to the cache configuration where it belongs.
So if travis is happy, :+1:

aertmann commented:
what's the reason for not caching this in development using the existing file monitoring on the node types files? https://github.com/neos/typo3cr/blob/master/Classes/TYPO3/TYPO3CR/Package.php#L49 + https://github.com/neos/neos/blob/master/Classes/TYPO3/Neos/Package.php#L43-L45

kitsunet commented:
I wasn't sure if it's actually worth the trouble, because changing a PostProcessor should then also lead to cache flush etc... But sure could be done.
Will (finally) take care of the merge error, sorry.

aertmann commented:
that's true, but they aren't monitored so that's harder to do and more understandable that it doesn't happen automatically.. but well it depends on how much this is of an improvement, but faster development setup is always preferred :)

kitsunet commented:
I would for now not cache this in development. I think it can easily lead to WTF moments. The improvement is nice, I will produce some numbers, but I think we can add caching for development in a separate change if we want to.

aertmann commented:
ok :+1: by reading

dfeyer commented:
:+1:
